### PR TITLE
Automatically load imported files in solc.

### PR DESCRIPTION
**Pull request by chriseth**
The same still needs to be done for the javascript compiler.

### CODE REVIEW

**Comment by VoR0220** [comment]
I'm not certain I understand what it is this is doing differently?

**Comment by VoR0220** [comment]
in OS X when trying to compile this I get this error

```
/Users/rcatalano/webthree-umbrella/solidity/libsolidity/interface/CompilerStack.cpp:400:24: error: call to 'move' is ambiguous
                                m_errors.push_back(move(err));
                                                   ^~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:1622:1: note: candidate function
      [with _Tp = std::__1::shared_ptr<dev::solidity::Error> &]
move(_Tp&& __t) _NOEXCEPT
^
/usr/local/include/boost/move/utility_core.hpp:211:77: note: candidate function [with T = std::__1::shared_ptr<dev::solidity::Error> &]
         inline typename ::boost::move_detail::remove_reference<T>::type && move(T&& t) BOOST_NOEXCEPT
                                                                            ^
[ 40%] Building CXX object libsolidity/CMakeFiles/solidity.dir/interface/SourceReferenceFormatter.cpp.o
1 error generated.
make[2]: *** [libsolidity/CMakeFiles/solidity.dir/interface/CompilerStack.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [libsolidity/CMakeFiles/solidity.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by chriseth** [comment]
The difference is that it now loads files automatically opon request by "import". Previously, it only read all the files that were supplied on the commandline and create an error if something else was imported.

**Comment by VoR0220** [comment]
Still may have a problem with the aliasing.

**Comment by VoR0220** [comment]
but that's a different topic. For now this should be merged.

File: libsolidity/interface/CompilerStack.cpp [hunk 1]
```diff
@@ -364,13 +375,44 @@ tuple<int, int, int, int> CompilerStack::positionFromSourceLocation(SourceLocati
...
+				errorMessage = "File not supplied initially.";
+			else
+				tie(contents, errorMessage) = m_readFile(path);
+			if (!errorMessage.empty())
+			{
+				auto err = make_shared<Error>(Error::Type::ParserError);
+				*err <<
+					errinfo_sourceLocation(import->location()) <<
+					errinfo_comment("Source not found: " + errorMessage);
+				m_errors.push_back(std::move(err));
```
Comment (chriseth) [inline libsolidity/interface/CompilerStack.cpp:400]: Actually I think this should be reported to the caller.

Comment (chriseth) [inline libsolidity/interface/CompilerStack.cpp:400]: It's fine, the compiler checks the error list after parsing.

**Pull request merged**